x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Authentication.java

```diff
@@ -447,11 +466,25 @@ public void writeToContext(ThreadContext ctx) throws IOException, IllegalArgumen
     }
 
     public String encode() throws IOException {
+        return Base64.getEncoder().encodeToString(BytesReference.toBytes(toVersionedBytes()));
+    }
+
+    public BytesReference toVersionedBytes() throws IOException {
```

Comment:
I'm blanking on a better name here (same for `fromVersionedBytes` below) but didn't want that to block the PR.

Reply:
I wanted to avoid the unnecessary base64 encoding for efficiency reasons (both the operation itself, and the slightly larger metadata blob). This is not critical though. In the interest of keeping things simple, I'll just re-use `encode` 👍